pkg/machines/libvirt.es6

```diff
@@ -299,9 +336,55 @@ LIBVIRT_PROVIDER = {
             failHandler: buildFailHandler({ dispatch, name, connectionName, message: _("VM SEND Non-Maskable Interrrupt action failed")}),
             args: ['inject-nmi', name]
         });
+    },
+
+    GET_HYPERVISOR_MAX_VCPU ({ connectionName }) {
+        logDebug(`${this.name}.GET_HYPERVISOR_MAX_VCPU:`);
+        if (connectionName) {
+            return dispatch => spawnVirsh({connectionName,
+                method: 'GET_HYPERVISOR_MAX_VCPU',
+                failHandler: buildFailHandler({ dispatch, connectionName, message: _("GET HYPERVISOR MAX VCPU action failed")}),
+                args: ['maxvcpus']
+            }).then((count) => dispatch(setHypervisorMaxVCPU(count)));
+        }
+
+        return unknownConnectionName(getHypervisorMaxVCPU);
     }
 };
 
+function unknownConnectionName (action) {
+    return  dispatch => {
+        return cockpit.user().done( loggedUser => {
+            const promises = Object.getOwnPropertyNames(VMS_CONFIG.Virsh.connections)
+                .filter(
+                    // The 'root' user does not have its own qemu:///session just qemu:///system
+                    // https://bugzilla.redhat.com/show_bug.cgi?id=1045069
+                    connectionName => canLoggedUserConnectSession(connectionName, loggedUser))
+                .map(connectionName => dispatch(action(connectionName)));
+
+            return cockpit.all(promises);
+        });
+    };    
+}
+
+function createTempFile (content) {
+    const dfd = cockpit.defer();
+    spawnScript({
+        script: "mktemp /tmp/abc-script.XXXXXX"
+    }).then(tempFilename => {
+        spawnScript({
+            script: `echo '${content}' > ${tempFilename}`
+        }).then(() => {
```

Comment:
This should use [cockpit.file](http://cockpit-project.org/guide/latest/cockpit-file.html) to avoid a shell and potential breakage when there are `'` in content.